lib/core/src/buy.rs

```diff
@@ -0,0 +1,65 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use sdk_common::prelude::{BreezServer, BuyBitcoinProviderApi, MoonpayProvider};
+
+use crate::{
+    model::{BuyBitcoinProvider, ChainSwap, Config},
+    prelude::LiquidNetwork,
+};
+
+#[async_trait]
+pub(crate) trait BuyBitcoinApi: Send + Sync {
+    /// Initiate buying Bitcoin and return a URL to the selected third party provider
+    async fn buy_bitcoin(
+        &self,
+        provider: BuyBitcoinProvider,
+        chain_swap: &ChainSwap,
+        redirect_url: Option<String>,
+    ) -> Result<String>;
+}
+
+pub(crate) struct BuyBitcoinService {
+    config: Config,
+    moonpay_provider: Arc<dyn BuyBitcoinProviderApi>,
+}
+
+impl BuyBitcoinService {
+    pub fn new(config: Config, breez_server: Arc<BreezServer>) -> Self {
+        let moonpay_provider = Arc::new(MoonpayProvider::new(breez_server));
+        Self {
+            config,
+            moonpay_provider,
+        }
+    }
+}
+
+#[async_trait]
+impl BuyBitcoinApi for BuyBitcoinService {
+    async fn buy_bitcoin(
+        &self,
+        provider: BuyBitcoinProvider,
+        chain_swap: &ChainSwap,
+        redirect_url: Option<String>,
+    ) -> Result<String> {
+        if self.config.network != LiquidNetwork::Mainnet {
+            return Err(anyhow!("Can only buy bitcoin on Mainnet"));
+        }
```

Comment:
Looks like Moonpay supports Testnet too: https://dev.moonpay.com/docs/sandbox-testing

> Once you've hit the "Pay" button, all Cryptocurrency withdrawals are executed on test networks:
>...
> Testnet3 for Bitcoin

Would that be easy to integrate?

Reply:
I'll add that

Reply:
@ok300 It's currently not possible to sign the Moonpay url using the staging breez server / sandbox secret. So I'll make an issue to support that

Reply:
Sounds good, thanks.